.github/workflows/code.yml

```diff
@@ -77,6 +77,10 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Set fake version if untagged
+        if: github.event_name != 'push' || !startsWith(github.ref, 'refs/tags')
+        run: echo "PRETEND_VERSION=$(git describe --tags | cut -d '-' -f 1,2 --output-delimiter '.dev')" >> $GITHUB_ENV
```

Comment:
What is the actual PRETEND_VERSION here? Will it change with every commit? If not, then will TestPyPI reject us uploading the same version number again?

Reply:
It will be that of our previous version, though I have since moved to appending seconds past epoch e.g. `0.4.0.123456798`